server/src/main/java/org/opensearch/cluster/metadata/ResourceLimitGroup.java

```diff
@@ -0,0 +1,269 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.metadata;
+
+import org.opensearch.cluster.AbstractDiffable;
+import org.opensearch.cluster.Diff;
+import org.opensearch.common.annotation.ExperimentalApi;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+import org.opensearch.core.xcontent.ToXContentObject;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.core.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Class to define the ResourceLimitGroup schema
+ * {
+ *     "analytics" : {
+ *              "jvm": 0.4,
+ *              "mode": "enforced",
+ *              "_id": "fafjafjkaf9ag8a9ga9g7ag0aagaga"
+ *      }
+ * }
+ */
+@ExperimentalApi
+public class ResourceLimitGroup extends AbstractDiffable<ResourceLimitGroup> implements ToXContentObject {
```

Comment:
Considering `name` below signifies a tenant, putting this inside ResourceLimitGroup looks confusing. I think it help by decoupling tenant and resource logic like below?


```
class Tenant {
   private String tenantName;
   private String tenantId;
   private List<ResourceLimitGroup> resourceLimitGroups; 
   // Can be extended with more details later if needed like tenant attributes etc. 
}
```

```
class ResourceLimitGroup {
    String resourceName;
    ResourceLimitGroupMode mode;
    Double threshold;
    // etc etc
 }
```
Let me know what you think.

Reply:
It makes sense but based on our internal discussion I think `QueryGroup` makes more sense here